### ISSUE TITLE
Watir::Capabilities class

### ISSUE THREAD

**Issue by titusfortner** [watir/watir#436]
Would it make sense to have Watir wrap the Selenium Remote::Capabilities class?

As I'm writing up documentation on how to set marionette: false for the upcoming Watir 6.0, this is what needs to be done:

```
caps = Selenium::WebDriver::Remote::Capabilities.firefox(marionette: false)
Selenium::WebDriver::Firefox::Binary.path = "/path/to/desired/firefox"
Watir::Browser.new desired_capabilities: caps
```

It seems like it would be much nicer to be able to do:

```
Watir::Browser.new(marionette: false, firefox_binary: "/path/to/binary")
```

But maybe we need to figure out the right approach on the Selenium side first, because I hate how we combine allowing both first class parameters & desired_capabilities parameters.

**Comment by ambirag** [comment]
@titusfortner 
just to add to what you are trying to do, currently we also need another way to pass Firefox profile, args and prefs for the new moz:firefoxOptions of marionette , if anyone is using selenium webdriver 3.\* series. afaik, this is not supported in watir, please let me know otherwise.

**Comment by titusfortner** [comment]
@ambirag - it is supported in Watir. For Firefox with geckodriver, Selenium 3 converts `:firefox_options` to `"moz:firefoxOptions"` before creating the new session. Let me know if you are finding issues.

**Comment by ambirag** [comment]
@titusfortner thanks, let me try and get back to you

**Comment by ambirag** [comment]
if i pass the "profile" in profile = Selenium::WebDriver::Firefox::Profile.new
    profile['accept_untrusted_certs']=true
    profile['assume_untrusted_certificate_issuer']=false
in 
capabilities = Selenium::WebDriver::Remote::Capabilities.firefox(marionette: true,firefox_options: {profile: "profile"}) , i get "Failed to unzip profile"
Is there any example how to create base64 encoded profile directory?

**Comment by ambirag** [comment]
@titusfortner

**Comment by titusfortner** [comment]
This was implemented with #583
